src/hw/rom/boot.c

```diff
@@ -1,12 +1,19 @@
 #include <stdio.h>
 #include "hw/rom/boot.h"
+#include "core/md5.h"
 #include "core/option.h"
 #include "hw/dreamcast.h"
 
 DEFINE_OPTION_STRING(bios, "dc_boot.bin", "Path to boot rom");
 
 #define BIOS_SIZE 0x00200000
 
+/* Known valid bios md5's */
+#define BIOS_CHINESE_MD5 "a5c6a00818f97c5e3e91569ee22416dc"
```

Comment:
I think we could throw these in an array, e.g.:
```
static const char *bios_md5[] = {
   "a5c6a00818f97c5e3e91569ee22416dc", /* chinese bios */
   ...
};
```

then make the validate method just loop over the array, returning 1 if any match.